### PR TITLE
Adding current user to application state

### PR DESCRIPTION
**Pull request by erickpintor**
I imagine this is the beginning of our user management system at the dashboard level.

For now I'm faking user auth with local storage api. We'll have to replace this with real authentication once the work on the website is complete.

I'm planing to reflect on the user type to enable intercom widget on a next PR.

### CODE REVIEW

File: src/authentication/recognizeUser.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { UnknownUser, CloudUser } from "./User"
+
+export const LAST_AUTH = "lastAuthenticatedUser"
+
+export default function recognizeUser(localStorage) {
```
Comment (erickpintor) [inline src/authentication/recognizeUser.js]: This function will have to be replaced with real authentication.

File: src/authentication/login.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    type: Actions.LOGOUT
+  }
+}
+
+const DEV_MODE = process.env.NODE_ENV === "development"
+
+export function reduceAuthentication(state = null, action) {
+  switch (action.type) {
+    case Actions.LOGIN:
+      if (DEV_MODE) localStorage.set(LAST_AUTH, action.user)
```
Comment (erickpintor) [inline src/authentication/login.js]: Handy during development but, insecure on cloud. We'll figure out what to do once real authentication is in place.

File: src/container/Container.js [hunk 3]
```diff
@@ -1,62 +1,55 @@
...
     this.bumpSchema = this.bumpSchema.bind(this);
     this.updateSecret = this.updateSecret.bind(this);
     this.observerCallback = this.observerCallback.bind(this);
     this.logout = this.logout.bind(this);
     this._onBreadcrumbItemClicked = this._onBreadcrumbItemClicked.bind(this);
   }
 
+  componentWillReceiveProps(next) {
+    if (this.props.currentUser === next.currentUser) return;
+    this.setState({ ...this.state, client: this._connect(next.currentUser) })
```
Comment (marrony) [inline src/container/Container.js]: the spread operator here is not necessary as react will update and not set the state

File: src/container/Container.js [hunk 4]
```diff
@@ -195,3 +187,12 @@ export default class Container extends Component {
     )
   }
 }
+
+function connectContainer(state, props) {
+  return {
+    ...props,
```
Comment (marrony) [inline src/container/Container.js]: same here, spread operator is not necessary as react will merge the properties returned by this function with the ones passed in jsx

**Comment by jchris** [comment]
Agree about the spread operator, aside from that LGTM

File: src/container/Container.js (hunk 3 above)
Comment (erickpintor) [inline src/container/Container.js]: What a great name they gave to `setState` method hum? :P

**Comment by marrony** [comment]
ship it

**Pull request merged**